client/packages/inventory/src/Stocktake/DetailView/ChangeLocationModal.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useState } from 'react';
+import {
+  InputWithLabelRow,
+  useTranslation,
+  ConfirmationWithChildrenModal,
+} from '@openmsupply-client/common';
+import {
+  LocationRowFragment,
+  LocationSearchInput,
+} from '@openmsupply-client/system';
+import { useStocktake } from '../api';
+
+interface ChangeLocationConfirmationModalProps {
+  isOpen: boolean;
+  onCancel: () => void;
+}
+
+export const ChangeLocationConfirmationModal = ({
+  isOpen,
+  onCancel,
+}: ChangeLocationConfirmationModalProps) => {
+  const t = useTranslation('inventory');
+
+  const [location, setLocation] = useState<LocationRowFragment | null>(null);
+
+  const onChangeLocation = useStocktake.line.changeLocation();
+
+  return (
+    <ConfirmationWithChildrenModal
+      isOpen={isOpen}
+      title={t('heading.are-you-sure')}
+      message={t('messages.confirm-change-location')}
+      onClose={onCancel}
+      onSave={() => onChangeLocation(location)}
```

Comment:
Unsure about UX here... currently the OK button is always enabled, so if you open the modal and then immediately press OK, it would clear the location for all lines. Would probably be better if they had to explicitly select "No location" or something right?

Reply:
Actually we have a component like that right ? I think something would need to be selected